helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixDataAccessor.java

```diff
@@ -599,4 +610,12 @@ private int constructOptions(PropertyType type) {
       List<DataUpdater<ZNRecord>> updaters, int options) {
     return _baseDataAccessor.updateChildren(paths, updaters, options);
   }
+
+  private void validateBucketizedEnabled(String path, boolean isRead) {
+    if (!BUCKETIZE_ZNRECORD_ENABLED) {
+      throw new HelixMetaDataAccessException(
+          "Can't " + (isRead ? "read" : "write") + " bucktized ZNode " + path
+              + " because Bucktize feature is not enabled. Please check ZK system property 'zk.bucketize.znrecord.enabled' .");
```

Comment:
Use `ZkSystemPropertyKeys.ZK_BUCKETIZE_ZNRECORD_ENABLED` instead of hard coding it?

Reply:
Are you suggesting we read the config every time? We could if the config need to be dynamically changed.

Reply:
Not reading the config every time. We use the config key to replace the hard coded string. The benefit is, it'll be easier for us to find the reference that uses the key. And if we refactor the name, we won't miss this place.
And actually the compiler usually will replace the constant variable "ZkSystemPropertyKeys.ZK_BUCKETIZE_ZNRECORD_ENABLED" with the string "zk.bucketize.znrecord.enabled".